examples/knative/powershell/kn-ps-vrni-databus/vrni-functions.ps1

```diff
@@ -0,0 +1,77 @@
+Import-Module CloudEvents.Sdk
+
+
+function New-vRNICloudEventsFromDatabus {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory = $true)]$Data,
+        [Parameter(Mandatory = $true)]$cloudEventSource
+    )
+
+    # empty container for the cloud events. vRNI can send multiple events in a single request, so we need to split it up
+    $cloudEvents = @()
+    $jsonData = $Data | ConvertFrom-Json -AsHashtable
+
+    # see whether the payload body has problems or application updates and process them differently
+    # first up: problems
+    if ($jsonData.ContainsKey("EntityMessageList")) {
+        # loop through all problems and create CloudEvents from them to give back
+        # to the main function (which fires them into the event broker)
+        foreach ($problemEvent in $jsonData["EntityMessageList"]) {
+
+            if (!$problemEvent.ContainsKey("entity")) {
+                Write-host "$(Get-Date) - Wrong format for problem event, skipping:`n"
+                Write-host $problemEvent
+                continue
+            }
+
+            # use this as the event type
+            # the payload has a "status" field with "UPDATE", "OPEN", or "CLOSE" to figure out what the problem update is about
+            $cloudEventType = "com.vmware.event.router/vrni.problem.v0"
+
+            try {
+                $cloudEvent = New-CloudEvent -Type $cloudEventType -Source $cloudEventSource -Id (New-Guid).Guid -Time (Get-Date)
```

Comment:
question: is there a stable and unique ID we can extract from the incoming event that we could use instead of creating new UUID? the benefit of it would be correlation and idempotency logic.

same for time btw.

Reply:
There's no trustworthy field in the payload as of now. This will change in the future, after which this can be updated.